core/src/com/unciv/logic/automation/BarbarianAutomation.kt

```diff
@@ -0,0 +1,224 @@
+            } else {
+                // heal
+                unit.fortifyIfCan()
+            }
+            return
+        }
+
+        // 2 - trying to upgrade
+        if (tryUpgradeUnit(unit, unitActions)) return
+
+        // 3 - trying to attack enemy
+        // if a embarked melee unit can land and attack next turn, do not attack from water.
+        if (battleHelper.tryDisembarkUnitToAttackPosition(unit, unitDistanceToTiles)) return
+        if (battleHelper.tryAttackNearbyEnemy(unit)) return
+
+        // 4 - trying to pillage tile or route
+        if (tryPillageImprovement(unit, unitDistanceToTiles, unitActions)) return
+
+        // 5 - heal the unit if needed
+        if (unit.health < 100) {
+            healUnit(unit, unitDistanceToTiles)
+            return
+        }
+
+        // 6 - wander
+        wander(unit, unitDistanceToTiles)
+    }
+
+    private fun automateScout(unit: MapUnit) {
+        val unitActions = UnitActions().getUnitActions(unit, UncivGame.Current.worldScreen)
+        val unitDistanceToTiles = unit.movement.getDistanceToTiles()
+        val nearEnemyTiles = battleHelper.getAttackableEnemies(unit, unitDistanceToTiles)
+
+        // 1 - heal or run if death is near
+        if (unit.health < 50) {
+            if (nearEnemyTiles.isNotEmpty()) {
+                // run
+                val furthestTile = findFurthestTile(unit, unitDistanceToTiles, nearEnemyTiles)
+                unit.movement.moveToTile(furthestTile)
+            } else {
+                // heal
+                unit.fortifyIfCan()
+            }
+            return
+        }
+
+        // 2 - trying to capture someone
+        // TODO
+
+        // 3 - trying to pillage tile or trade route
+        if (tryPillageImprovement(unit, unitDistanceToTiles, unitActions)) return
+
+        // 4 - heal the unit if needed
+        if (unit.health < 100) {
+            healUnit(unit, unitDistanceToTiles)
+            return
+        }
+
+        // 5 - wander
+        wander(unit, unitDistanceToTiles)
+    }
+
+    private fun findFurthestTile(
+            unit: MapUnit,
+            unitDistanceToTiles: PathsToTilesWithinTurn,
+            nearEnemyTiles: List<AttackableTile>
+    ): TileInfo {
+        val possibleTiles = unitDistanceToTiles.keys.filter { unit.movement.canMoveTo(it) }
+        val enemies = nearEnemyTiles.mapNotNull { it.tileToAttack.militaryUnit }
+        var furthestTile: Pair<TileInfo, Float> = possibleTiles.random() to 0f
+        for (enemy in enemies) {
+            for (tile in possibleTiles) {
+                val distance = enemy.movement.getMovementCostBetweenAdjacentTiles(enemy.currentTile, tile, enemy.civInfo)
+                if (distance > furthestTile.second) {
+                    furthestTile = tile to distance
+                }
+            }
+        }
+        return furthestTile.first
+    }
+
+    private fun healUnit(unit: MapUnit, unitDistanceToTiles: PathsToTilesWithinTurn) {
```

Comment:
Since this function already exists in UnitAutomation, why not use that directly? Is there a difference between the two?

Reply:
this function quite different